### PR TITLE
[Core] Refactor default payment method resolver

### PR DESCRIPTION
**Pull request by Zales0123**
| Q               | A
| --------------- | ---
| Bug fix?        | yes |
| New feature?    | no |
| BC breaks?      | yes |
| Related tickets | |
| License         | MIT |

`DefaultPaymentMethodResolver` from *Payment* component is not used in *Sylius*, better - it's not even registered. Is there any reason to share interface between this class and `DefaultPaymentMethodResolver` from *Core*? It only forces us to do some weird stuff like passing `Payment` to resolver, necessarily with set `Order` to get channel from it rather than just passing an `Order` object.

### CODE REVIEW

File: src/Sylius/Component/Core/Resolver/DefaultPaymentMethodResolverInterface.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+interface DefaultPaymentMethodResolverInterface
+{
+    /**
+     * @param ChannelInterface $channel
+     *
+     * @return PaymentMethodInterface
+     */
+    public function getDefaultPaymentMethodByChannel(ChannelInterface $channel);
```
Comment (pamil) [inline src/Sylius/Component/Core/Resolver/DefaultPaymentMethodResolverInterface.php]: `getForChannel(ChannelInterface $channel)`?

File: src/Sylius/Component/Core/Resolver/DefaultPaymentMethodResolverInterface.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+interface DefaultPaymentMethodResolverInterface
+{
+    /**
+     * @param ChannelInterface $channel
+     *
+     * @return PaymentMethodInterface
```
Comment (pamil) [inline src/Sylius/Component/Core/Resolver/DefaultPaymentMethodResolverInterface.php]: + `@throws Unreso...`

**Comment by pjedrzejewski** [comment]
ByChannel sounds like a concrete implementation, I preferred the previous interface, which was more generic. You may want to resolve payment methods based on something else than channel. (from order)

**Comment by michalmarcinkowski** [comment]
Agree with @pjedrzejewski. The concept of this resolver is the following: "give me a default payment method for this payment" and in the specific implementation we can resolve the payment method based on whatever we want. The same concept works for shipping and provide good abstraction to any requirements.

**Comment by Zales0123** [comment]
Ok, I just understood this as "give me default payment method" and that's it - that's why it was weird to me to pass payment to this method, as it looked for like some workaround ;) Closing then.

**Pull request closed without merging**